src/continuous_agg.c

```diff
@@ -1242,28 +1489,78 @@ ts_continuous_agg_watermark(PG_FUNCTION_ARGS)
 	PG_RETURN_INT64(watermark->value);
 }
 
-/*
- * Determines bucket width for given continuous aggregate.
- *
- * Currently all buckets are fixed in size but this is going to change in the
- * future. Any code that needs to know bucket width has to determine it using
- * this procedure instead of accessing any ContinuousAgg fields directly.
- */
+/* Determines if bucket width if variable for given continuous aggregate. */
+bool
+ts_continuous_agg_bucket_width_variable(const ContinuousAgg *agg)
+{
+	return agg->data.bucket_width == BUCKET_WIDTH_VARIABLE;
+}
+
+/* Determines bucket width for given continuous aggregate. */
 int64
 ts_continuous_agg_bucket_width(const ContinuousAgg *agg)
 {
+	if (ts_continuous_agg_bucket_width_variable(agg))
+	{
+		/* should never happen, this code is useful mostly for debugging purposes */
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("bucket width is not defined for a variable bucket")));
+	}
+
 	return agg->data.bucket_width;
 }
 
-/*
- * Determines maximum possible bucket width for given continuous aggregate.
- *
- * E.g. for monthly continuous aggreagtes this procedure will return 31 days.
- * This is not true for ts_continuous_agg_bucket_width which may use additional
- * arguments to determine the width of a concrete bucket.
- */
+/* Determines maximum possible bucket width for given continuous aggregate. */
 int64
 ts_continuous_agg_max_bucket_width(const ContinuousAgg *agg)
```

Comment:
The maximum widths were introduced by https://github.com/timescale/timescaledb/commit/3f9adffaf0a74d8d33a3de9f2994f86ee1c9bdf5

If I understand correctly we are abandoning them. In that case I think you should get rid of them completely.

Reply:
max_bucket_width existed long before 3f9adff, see CaggsInfo and CaggInvalidationState structures. In 3f9adff the access to max_bucket_with was only wrapped in a procedure, which was very helpful during developing this feature.

You are right that we don't need max_bucket_width because it always equals bucket_with. The first thing I'm going to do after merging this PR is to submit a corresponding refactoring. I have a corresponding deliverable in the epic task https://github.com/timescale/timescaledb-private/issues/926

Reply:
`CaggsInfo` was introduced quite recently and `CaggInvalidationState` does not have something either, look here:
https://github.com/timescale/timescaledb/blob/3f9adffaf0a74d8d33a3de9f2994f86ee1c9bdf5/tsl/src/continuous_aggs/invalidation.c#L80

The above is a snapshot of https://github.com/timescale/timescaledb/commit/3f9adffaf0a74d8d33a3de9f2994f86ee1c9bdf5
and there is nothing in `CaggInvalidationState`

Reply:
Hmmm... you are right, it looks like we overlooked adding max_bucket_width to CaggInvalidationState during the code review of 221437e8. There was no need for it. Anyway, I will refactor the code and get rid entirely of max_bucket_width after merging this PR.